### DISCUSSION TITLE
Dependency graph for custom private dependencies

### DISCUSSION THREAD

**Discussion by LeKristapino** [community/community#36638] [Code Security]
Hi,

I spent about half a day (first action) making use of the dependency submission API to push the dependencies of our private repos, that we use for our C++ projects. 

After successfully making this work, I realized that this functionality is irrelevant because there are zero actions that I can take or utilize this information in any way.

Basically:

* Our dependencies are listed in a custom CMake file, so viewing dependencies and versions is as easy as opening a file in the root of the repository (so no visual improvement in terms of just checking out dependencies of the repo)

* Dependabot is not working (no way to make a "custom" dependabot to make auto PRs on new version availability)
* No references with URLs to the actual GitHub repos of the dependencies (this I can somewhat understand with regards to security, but still)
* There is no way how to get the current dependency graph of a repo through API so it doesn't help in making a custom dependency updater in any way

All of my efforts were done in hopes of making a custom dependabot-like functionality where I could link all the dependencies through this functionality and then automatically trigger PR when one of those dependencies has a new version.

What I would expect more with regard to the dependency graph:

1) Actual linking between dependency repos and the master repo (even if it is only under the hood).
2) Possibility to create a custom action that will use before mentioned links to trigger a PR in all repos that depend on the updated dependency when a new version comes out (it could be made from either end - scheduled action on the master repo to check for dependency updates also seems ok)
3) Possibility to get the current dependency graph of a repo through API (not just compare changes between commits) - even if the first two suggestions are impossible or hard to make - this would allow creating easy serverless functions or HTTP triggers that could create before mentioned PRs through API
4) Some more flexibility with dependency graph in the scope of an organization  

Basically, my main hope was that in utilizing dependency graph, I could make custom auto-triggered dependency updates on my repos, but it seems that at its current stage, dependency graph functionality  (unless you use supported dep managers) is useless
